protocols/gossipsub/src/config.rs

```diff
@@ -183,13 +182,6 @@ impl Config {
         self.protocol.max_transmit_size
     }
 
-    /// The time a connection is maintained to a peer without being in the mesh and without
-    /// send/receiving a message from. Connections that idle beyond this timeout are disconnected.
-    /// Default is 120 seconds.
-    pub fn idle_timeout(&self) -> Duration {
```

Comment:
Removing an option is a breaking change so we will have to bump the minor version instead of patch!

We don't merge breaking changes immediately but batch them up, see the current milestone (https://github.com/libp2p/rust-libp2p/milestone/7). This allows us to better plan, when we release a breaking change to our users and do it less frequently!

Also, to make the transition easier for our users, it would be great to first deprecate this config option and point users towards the new `SwarmBuilder::idle_connection_timeout` function. See https://github.com/libp2p/rust-libp2p/pull/4029. Does that make sense? :)

Concretely, I'd suggest:

- Update this PR to be a breaking change by bumping the version to `0.46.0`. We will then add it to the above milestone.
- Optionally: Send another PR where we just place `#[deprecated]` on this function and immediately release that to users.

Reply:
Roger that. I opened another PR #4648 for adding deprecated on function `Config::idle_timeout`.

Reply:
Awesome, thank you!